From c903b48af88aa0a6eb61493f81f6cf6d6aa6738f Mon Sep 17 00:00:00 2001
From: Yiwei Zhang <zzyiwei@google.com>
Date: Sat, 15 Aug 2020 13:48:36 -0700
Subject: [PATCH 2/4] vulkan: avoid waiting for service.sf.present_timestamp
 creation

This change is to help unblock:
1. ANGLE used as built-in gl driver
2. Vulkan used as RenderEngine backend

Moving the property creation before RenderEngine init can potentially
cause a boot time regression. Those creating gpu contexts before boot
finished are not relying on EGL_ANDROID_get_frame_timestamps or
VK_GOOGLE_display_timing.

Bug: 164717293
Test: manual
Change-Id: Iff34e0a5c4101c3961a078e0d34a2a7fb0391857
---
 vulkan/libvulkan/driver.cpp | 4 +---
 1 file changed, 1 insertion(+), 3 deletions(-)

diff --git a/vulkan/libvulkan/driver.cpp b/vulkan/libvulkan/driver.cpp
index f840561..79ff17a 100644
--- a/vulkan/libvulkan/driver.cpp
+++ b/vulkan/libvulkan/driver.cpp
@@ -956,9 +956,7 @@ VkResult EnumerateDeviceExtensionProperties(
 
     // conditionally add VK_GOOGLE_display_timing if present timestamps are
     // supported by the driver:
-    const std::string timestamp_property("service.sf.present_timestamp");
-    android::base::WaitForPropertyCreation(timestamp_property);
-    if (android::base::GetBoolProperty(timestamp_property, true)) {
+    if (android::base::GetBoolProperty("service.sf.present_timestamp", false)) {
         loader_extensions.push_back({
                 VK_GOOGLE_DISPLAY_TIMING_EXTENSION_NAME,
                 VK_GOOGLE_DISPLAY_TIMING_SPEC_VERSION});
-- 
2.32.0

